### PR TITLE
Implement authentication for the web UI

### PR DESCRIPTION
**Pull request by adejanovski**
Based on the work made in this PR : https://github.com/spotify/cassandra-reaper/pull/146

### CODE REVIEW

File: src/server/src/main/java/io/cassandrareaper/ReaperApplicationConfiguration.java [hunk 1]
```diff
@@ -434,4 +450,18 @@ public String toString() {
     /* Each datacenter requires at minimum one reaper instance that has jmx access to all nodes in that datacenter */
     EACH
   }
+
+  public static class AccessControlConfiguration {
```
Comment (michaelsembwever) [inline src/server/src/main/java/io/cassandrareaper/ReaperApplicationConfiguration.java]: nit-pick: can the inner class be `final`?

File: src/server/src/main/java/io/cassandrareaper/resources/LoginResource.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import com.google.common.collect.Maps;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.shiro.authc.AuthenticationException;
+import org.apache.shiro.authc.IncorrectCredentialsException;
+import org.apache.shiro.authc.UsernamePasswordToken;
+import org.apache.shiro.subject.Subject;
+import org.secnod.shiro.jaxrs.Auth;
+
+@Path("/")
+public class LoginResource {
```
Comment (michaelsembwever) [inline src/server/src/main/java/io/cassandrareaper/resources/LoginResource.java]: can we put `LoginResource` into the `io.cassandrareaper.resources.auth` package, now that it exists and is "the feature" this belongs to?

**Review by michaelsembwever — APPROVED** [review]

File: src/server/src/main/java/io/cassandrareaper/ReaperApplicationConfiguration.java (hunk 1 above)
Comment (adejanovski) [reply src/server/src/main/java/io/cassandrareaper/ReaperApplicationConfiguration.java]: Sure thing

File: src/server/src/main/java/io/cassandrareaper/resources/LoginResource.java (hunk 2 above)
Comment (adejanovski) [reply src/server/src/main/java/io/cassandrareaper/resources/LoginResource.java]: Done

**Pull request merged**